classes/pdf/HTMLTemplateOrderSlip.php

```diff
@@ -67,11 +67,40 @@ public function __construct(OrderSlip $order_slip, Smarty $smarty)
         // header informations
         $this->date = Tools::displayDate($this->order_slip->date_add);
         $prefix = Configuration::get('PS_CREDIT_SLIP_PREFIX', Context::getContext()->language->id);
-        $this->title = sprintf(HTMLTemplateOrderSlip::l('%1$s%2$06d'), $prefix, (int) $this->order_slip->id);
+
+        $this->title = sprintf(
+            HTMLTemplateOrderSlip::l('%1$s%2$06d'),
+            $prefix,
+            Configuration::get('PS_CREDIT_SLIP_RESET') ? $this->getNumberOrderSlip((int) $order_slip->id) : (int) $order_slip->id
+        );
 
         $this->shop = new Shop((int) $this->order->id_shop);
     }
 
+    /**
+     * Returns the numbering of the subscription invoice with respect to the year.
+     *
+     * @param int $id_order_slip order_slip
+     *
+     * @return int number for year
+     */
+    public function getNumberOrderSlip(int $id_order_slip): int
+    {
+        $date = Db::getInstance()->getValue('SELECT date_add FROM `' . _DB_PREFIX_ . 'order_slip` WHERE id_order_slip=' . $id_order_slip);
+
+        if (isset($date)) {
+            $current_year = date('Y-01-01 00:00:00', strtotime($date));
+
+            $counts = Db::getInstance()->getValue(
+                'SELECT count(*) as "n" FROM `' . _DB_PREFIX_ . 'order_slip` WHERE date_add > "' . $current_year . '" AND date_add < "' . $date . '"'
```

Comment:
I don't understand why you add `as "n"`, also I also don't understand why you are trying to find order_slip not related to the id_order_slip you have in parameter and not related to the order itself. It returns all order_slip in the database :thinking:

Reply:
In this case, the 'as "n"' is no longer needed when using "getValue". I can leave it as "SELECT count (*) FROM` '", it was to give the column a value.

It would only retrieve the items that are between the selected ID and the calendar year to do a count.

![imagen](https://user-images.githubusercontent.com/30494301/116452183-b65e7280-a85d-11eb-8356-d856a93633be.png)

Reply:
You can remove the `"n"` which is useless